sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -82,36 +84,155 @@ private DoFnSignatures() {}
     return signature;
   }
 
+  /**
+   * The global context for the class, for use in analysis.
```

Comment:
"global" rubs me the wrong way; maybe call it "FnAnalysisContext" and fnContext respectively?